src/scripts/api/CraftingSystemDataExchanger.ts

```diff
@@ -0,0 +1,413 @@
+
+interface IngredientMasterCrafted {
+    id: string;
+    name?: string;
+    components?: ComponentMasterCrafted[];
+    recipe?: RecipeMasterCrafted;
+}
+
+interface ProductMasterCrafted {
+    id: string;
+    name: string;
+    components: ComponentMasterCrafted[];
+}
+
+interface ComponentMasterCrafted {
+    id?: string;
+    uuid?: string;
+    quantity?: number;
+    name?: string;
+}
+
+interface RecipeMasterCrafted {
+    id?: string;
+    name?: string;
+    description?: string;
+    time?: number;
+    macroName?: string;
+    recipeBook?: RecipeBookMasterCrafted;
+    ingredients?: IngredientMasterCrafted[];
+    products?: ProductMasterCrafted[];
+    tools?: string[];
+    sound?: string;
+    ownership?: Record<string, unknown>;
+    ingredientsInspection?: 0 | 1;
+    productInspection?: 0 | 1;
+    img?: string;
+}
+
+interface RecipeBookMasterCrafted {
+    id: string;
+    name: string;
+    description: string;
+    recipes?: RecipeMasterCrafted[];
+    tools?: string[];
+    sound?: string;
+    ownership?: Record<string, unknown>;
+    ingredientsInspection?: 0 | 1;
+    productInspection?: 0 | 1;
+    img: string;
+}
+
+export {IngredientMasterCrafted, ProductMasterCrafted, ComponentMasterCrafted, RecipeMasterCrafted, RecipeBookMasterCrafted};
+
+interface MasterCraftedDataExchanger extends CraftingSystemDataExchanger<RecipeBookMasterCrafted> {}
+
+export {MasterCraftedDataExchanger};
+
+class DefaultMasterCraftedDataExchanger implements MasterCraftedDataExchanger {
+
+    private readonly recipeAPI: RecipeAPI;
+    private readonly essenceAPI: EssenceAPI;
+    private readonly componentAPI: ComponentAPI;
+    private readonly craftingSystemAPI: CraftingSystemAPI;
+    private readonly localizationService: LocalizationService;
+    private readonly notificationService: NotificationService;
+
+    constructor({
+        recipeAPI,
+        essenceAPI,
+        componentAPI,
+        craftingSystemAPI,
+        localizationService,
+        notificationService,
+    }: {
+        recipeAPI: RecipeAPI;
+        essenceAPI: EssenceAPI;
+        componentAPI: ComponentAPI;
+        craftingSystemAPI: CraftingSystemAPI;
+        localizationService: LocalizationService;
+        notificationService: NotificationService;
+    }) {
+        this.recipeAPI = recipeAPI;
+        this.essenceAPI = essenceAPI;
+        this.componentAPI = componentAPI;
+        this.craftingSystemAPI = craftingSystemAPI;
+        this.localizationService = localizationService;
+        this.notificationService = notificationService;
+    }
+
+    get notifications(): NotificationService {
+        return this.notificationService;
+    }
+
+    async import(_importData: RecipeBookMasterCrafted): Promise<CraftingSystemData> {
+        throw new Error("Method not implemented.");
+    }
+
+    async export(_craftingSystemId: string): Promise<RecipeBookMasterCrafted> {
+        throw new Error("Method not implemented.");
+    }
```

Comment:
@p4535992 this is where we'd need to implement  the logic you started on in https://github.com/misterpotts/fabricate/pull/322

Reply:
Sorry for the delay i'll try out this weekend

Reply:
No worries! Whenever is good for you :)